src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoData.cs

```diff
@@ -41,10 +41,66 @@ public static string GetAbbreviatedEraName(CultureInfo cultureInfo)
             throw GetCultureNotSupportedException(cultureInfo);
         }
 
+        public static string GetLongDatePattern(CultureInfo cultureInfo)
+        {
+            if (string.Equals(cultureInfo.Name, "en-US", StringComparison.OrdinalIgnoreCase))
+            {
+                return s_isWindows ? "dddd, MMMM dd, yyyy" : "dddd, MMMM d, yyyy";
+            }
+            if (string.Equals(cultureInfo.Name, "es-ES", StringComparison.OrdinalIgnoreCase))
+            {
+                return s_isWindows ? "dddd, d' de 'MMMM' de 'yyyy" : "dddd, d 'de' MMMM 'de' yyyy";
+            }
+
+            throw GetCultureNotSupportedException(cultureInfo);
+        }
+
+        public static string GetMonthDayPattern(CultureInfo cultureInfo)
+        {
+            if (string.Equals(cultureInfo.Name, "en-US", StringComparison.OrdinalIgnoreCase))
+            {
+                return "MMMM d";
+            }
+            if (string.Equals(cultureInfo.Name, "es-ES", StringComparison.OrdinalIgnoreCase))
+            {
+                return s_isWindows ? "d' de 'MMMM" : "d 'de' MMMM";
+            }
+
+            throw GetCultureNotSupportedException(cultureInfo);
+        }
+
+        public static string GetShortDatePattern(CultureInfo cultureInfo)
+        {
+            if (string.Equals(cultureInfo.Name, "en-US", StringComparison.OrdinalIgnoreCase))
+            {
+                return "M/d/yyyy";
+            }
+            if (string.Equals(cultureInfo.Name, "es-ES", StringComparison.OrdinalIgnoreCase))
+            {
+                return s_isWindows ? "dd/MM/yyyy" : "d/M/yyyy";
+            }
+
+            throw GetCultureNotSupportedException(cultureInfo);
+        }
+
+        public static string GetYearMonthPattern(CultureInfo cultureInfo)
+        {
+            if (string.Equals(cultureInfo.Name, "en-US", StringComparison.OrdinalIgnoreCase))
```

Comment:
I don't like the idea of testing the passing culture name here. can we just pass all needed in the InlineData attribute? the current model will be hard to extend if we need to add more cultures. but if you used InlineData attribute to specify all needed data to test against that will make it extensible